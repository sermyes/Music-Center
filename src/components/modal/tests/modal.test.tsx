import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import Modal from '../modal';
import userEvent from '@testing-library/user-event';

describe('Modal', () => {
  const Child = () => <div>modal</div>;
  const onClose = jest.fn();

  it('renders correctly', () => {
    const component = renderer.create(
      <Modal onClose={onClose}>
        <Child />
      </Modal>
    );

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders modal', async () => {
    render(
      <Modal onClose={onClose}>
        <Child />
      </Modal>
    );
    const button = screen.getByRole('button');
    await userEvent.click(button);

    expect(screen.getByText('modal')).toBeInTheDocument();
    expect(onClose).toHaveBeenCalledTimes(1);
  });
});
