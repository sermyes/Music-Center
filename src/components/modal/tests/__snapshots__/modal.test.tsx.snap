// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal renders correctly 1`] = `
Array [
  <div
    className="overlay"
  />,
  <div
    className="wrapper"
  >
    <div
      className="inner"
    >
      <button
        className="iconBtn"
        onClick={[Function]}
      >
        <i
          className="icon fas fa-times"
        />
      </button>
      <div>
        modal
      </div>
    </div>
  </div>,
]
`;
